susemanager-utils/susemanager-sls/salt-ssh/preflight.sh

```diff
@@ -202,6 +202,21 @@ elif [ "${INSTALLER}" = "apt" ]; then
     fi
 fi
 
+SELINUX_POLICY_FILENAME="salt_ssh_port_forwarding.cil"
+function selinux_policy_loaded {
+    semodule -l | grep -x $SELINUX_POLICY_FILENAME
+}
+
+# Our SSH tunnel uses a custom port and we must configure SELinux to account for it
+if [[ $REPO_HOST == "localhost" ]] && command -v selinuxenabled && selinuxenabled; then
+    if ! selinux_policy_loaded; then
+        if ! [[ -f $SELINUX_POLICY_FILENAME ]] && [[ $REPO_PORT =~ ^[0-9]+$ ]]; then
+            echo "(portcon tcp ${REPO_PORT} (system_u object_r ssh_port_t ((s0)(s0))))" >$SELINUX_POLICY_FILENAME
```

Comment:
Also how many times do you expect to run this script on the same host? I suppose it will be ran only once at install time.

 Is there a value in checking that `$SELINUX_POLICY_FILENAME` does not already exist? Maybe you could unconditionally write the policy module, and _then_ verify that the file exists (and hard fail if it doesn't).

Reply:
Another thought. I assume this script is running as root, and is reading/writing files into root-controlled folders. Specifically, the CIL policy module file will be written where? In the `root` user home?

If this file could be pre-created by another user, you could be tricked into loading an arbitrary policy module. Even if you unconditionally overwrite the file every time, as I suggest in the previous comment, if the file is controlled by another user it could still be modified by winning a race between the time when you write it and when you use it.

Reply:
Having looked into it, the CIL policy module file will be written in `$TEMP_DIR`, which has been safely created ahead of time with `mktemp -d -t salt-bundle-XXXXXXXXXX`. This directory is created with permissions u+rwx, and thus its contents will be fresh and root-controlled.

Therefore there is no worry about policy module pre-creation, as I was reasoning in the previous comment. 

However, this also means that the policy file will never exist upon launching the script, since the temp folder will have been created anew. So there's no point in checking if the file already exists. Since I don't see `set -x` in the script, I would rather check that the `semodule -i` call has been successful, and hard fail if not.

Reply:
Generally, the script will be executed once. It's executed by "Salt SSH" and "Salt SSH" does not blindly execute it multiple times. But there are situations in which it will be executed again, and I wanted to be careful about those.

You raise another good point about the file location. I was not sure if I wanted to leave it in `/root/`, which would allow a) our users to check what we've allowed (but maybe there is a better way to do it, I don't know SELinux well) and b) might help us with debugging. But then I just implemented it for `$TEMP_DIR` and there a file exists check is indeed pointless.

Generally, when `semodule -i` fails, the next function call that tries to use the tunnel will also fail (that's how we got the bug report in the first place). But I agree with adding a check for `semanage -i`'s exit code, that's a better way to do it. 

I'll revisit this tomorrow again, thanks for your review!

Reply:
I removed the `-f` check, I think we don't really need to keep the file around necessarily, and added the check for `semanage -i`.